test/test_custom_gates.jl

```diff
@@ -169,4 +169,119 @@ using BraketSimulator: DoubleExcitation, SingleExcitation, matrix_rep, Control
+        end
+        @test qubit_count(DoubleExcitationMinus(ϕ)) == 4
+        @test inv(DoubleExcitationMinus(ϕ)) == DoubleExcitationMinus(-ϕ)
+    end
+
+    @testset "Double excitation plus" begin
+        ϕ  = 3.56
+        nq = 4
+        instructions = [Instruction(H(), [0]), Instruction(H(), [1]), Instruction(DoubleExcitationPlus(ϕ), [0, 1, 2, 3])]
+        # instructions for the matrix representation
+        u_instructions = [Instruction(H(), [0]), Instruction(H(), [1]), Instruction(Unitary(Matrix(matrix_rep(DoubleExcitationPlus(ϕ)))), [0, 1, 2, 3])]
+        state_vector =  0.5 * [exp(im*ϕ/2.0), 0, 0, -sin(ϕ/2), exp(im*ϕ/2.0), 0, 0, 0, exp(im*ϕ/2.0), 0, 0, 0, cos(ϕ/2), 0, 0, 0]
+        probability_amplitudes = [0.2499999999999999, 0.0, 0.0, 0.23921715039269298, 0.2499999999999999, 0.0, 0.0, 0.0, 0.2499999999999999, 0.0, 0.0, 0.0, 0.01078284960730691, 0.0, 0.0, 0.0]
+        @testset "Simulator $sim, instruction set $ix_label" for sim in (StateVectorSimulator, DensityMatrixSimulator),
+            (ix_label, ixs) in (("raw", instructions), ("unitary", u_instructions))
+            simulation = sim(nq, 0)
+            simulation = evolve!(simulation, ixs)
+            if sim == StateVectorSimulator
+                @test state_vector ≈ collect(BraketSimulator.state_vector(simulation))
+            end
+            @test probability_amplitudes ≈
+                  collect(BraketSimulator.probabilities(simulation))
+        end
+        @test qubit_count(DoubleExcitationPlus(ϕ)) == 4
+        @test inv(DoubleExcitationPlus(ϕ)) == DoubleExcitationPlus(-ϕ)
+    end
+
+    @testset "FermionicSWAP" begin
+        ϕ  = 3.56
+        nq = 2
+        instructions = [Instruction(H(), [0]), Instruction(H(), [1]), Instruction(FermionicSWAP(ϕ), [0, 1])]
+        # instructions for the matrix representation
+        u_instructions = [Instruction(H(), [0]), Instruction(H(), [1]), Instruction(Unitary(Matrix(matrix_rep(FermionicSWAP(ϕ)))), [1, 0])]
+        state_vector = 0.5 * [1, exp(im*ϕ/2.0)*cos(ϕ / 2.0) - im*exp(im*ϕ/2.0)*sin(ϕ/2.0), 
+                              - im*exp(im*ϕ/2.0)*sin(ϕ/2.0) + exp(im*ϕ/2.0)*cos(ϕ/2.0), exp(im * ϕ)]
+        probability_amplitudes = [0.2499999999999999, 0.2499999999999999, 0.2499999999999999, 0.2499999999999999]
+        @testset "Simulator $sim, instruction set $ix_label" for sim in (StateVectorSimulator, DensityMatrixSimulator),
+            (ix_label, ixs) in (("raw", instructions), ("unitary", u_instructions))
+            simulation = sim(nq, 0)
+            simulation = evolve!(simulation, ixs)
+            if sim == StateVectorSimulator
+                @test state_vector ≈ collect(BraketSimulator.state_vector(simulation))
+            end
+            @test probability_amplitudes ≈
+                  collect(BraketSimulator.probabilities(simulation))
+        end
+        @test qubit_count(FermionicSWAP(ϕ)) == 2
+        @test inv(FermionicSWAP(ϕ)) == FermionicSWAP(-ϕ)
```

Comment:
This (and the other `inv` tests) would also be good to check with a conversion to `Unitary` like:
```
@test inv(g) == inv(Unitary(matrix_rep(g))
```

Reply:
Please see this:

I tried this for SingleExcitation, seems method not defined for Unitary.
```
inv(SingleExcitation(3.56)) == inv(Unitary(matrix_rep(SingleExcitation(3.56))))


ERROR: MethodError: no method matching Unitary(::StaticArraysCore.SMatrix{4, 4, ComplexF64, 16})

Closest candidates are:
  Unitary(::Matrix{<:Number})
   @ Braket ~/Desktop/New/braket/4/Braket.jl/src/gates.jl:133
  Unitary(::Array{Array{Vector{T}, 1}, 1}) where T
   @ Braket ~/Desktop/New/braket/4/Braket.jl/src/gates.jl:135
  (::Type{G})(::Tuple{}) where G<:Gate
   @ Braket ~/Desktop/New/braket/4/Braket.jl/src/gates.jl:99

```